compass/ocean/tests/baroclinic_gyre/initial_state.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import xarray as xr
+from mpas_tools.cime.constants import constants
+from mpas_tools.io import write_netcdf
+
+from compass.ocean.vertical import init_vertical_coord
+from compass.step import Step
+
+
+class InitialState(Step):
+    """
+    A step for creating the initial condition for the baroclinic gyre
+    test cases
+
+    Attributes
+    ----------
+    resolution : float
+        The resolution of the test case (m)
+    """
+    def __init__(self, test_case, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        resolution : float
+            The resolution of the test case (m)
+        """
+        super().__init__(test_case=test_case, name='initial_state')
+        self.resolution = resolution
+        self.add_input_file(
+            filename='culled_mesh.nc',
+            target='../cull_mesh/culled_mesh.nc')
+        self.add_input_file(
+            filename='culled_graph.info',
+            target='../cull_mesh/culled_graph.info')
+
+        self.add_output_file('initial_state.nc')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        config = self.config
+
+        dsMesh = xr.open_dataset('culled_mesh.nc')
+
+        ds = _write_initial_state(config, dsMesh)
+
+        _write_forcing(config, ds.latCell)
+
+
+def _write_initial_state(config, dsMesh):
+
+    ds = dsMesh.copy()
+
+    bottom_depth = config.getfloat('vertical_grid', 'bottom_depth')
+    ds['bottomDepth'] = bottom_depth * xr.ones_like(ds.xCell)
+    ds['ssh'] = xr.zeros_like(ds.xCell)
+
+    init_vertical_coord(config, ds)
+
+    # setting the initial conditions
+    section = config['baroclinic_gyre']
+    initial_salinity = section.getfloat('initial_salinity')
+    temp_top = section.getfloat('initial_temp_top')
+    temp_bot = section.getfloat('initial_temp_bot')
+
+    print(f'zmid: {ds.zMid[0,0,:].values}')
+    cc = 0.049  # 0.049 value from optimization on 1800m
+    aa = temp_top / np.log(cc)
+    bb = (cc ** (temp_bot / temp_top) - cc)
+    temperature = aa * np.log(bb * -1.0 * ds.zMid / bottom_depth + cc)
+    print(f'Tinit: {temperature[0,0,:].values}')
+    val_bot = aa * np.log(bb + cc)
+    val_top = aa * np.log(cc)
+    print(f'analytical bottom T: {val_bot} at \
+            depth : {bottom_depth}')
```

Comment:
Here and in similar places, this is the preferred way to continue an f-string on a subsequent line:
```suggestion
    print(f'analytical bottom T: {val_bot} at '
            f'depth : {bottom_depth}')
```

Reply:
Thanks @xylar , I modified these. Let me know if I missed others.